### DISCUSSION TITLE
GraphQL API: getObjectListing returns an empty list

### DISCUSSION THREAD

**Discussion by Piotrulaa** [pimcore/pimcore#12265] [Q&A / Support] [answered]
Hi, I encountered a problem with retrieving list of my product DataObjects through DataHub GraphQL API. I can retrieve single product with getProduct query by providing its id, but when I try to retrieve more than one product using getProductListing the response is always empty list, no matter how I modify the query parameters.

Here is example of queries I made:
```
{
  getProduct(id: 2) {
    Title
  }
  
  getProductListing(first: 10) {
    totalCount
    edges {
      node {
        Title
      }
    }
  }
}
```

I will be grateful for any help with this problem.

**Comment by Piotrulaa** [comment]
Does anyone have any ideas? Please, I'm really stuck with this one.

**Reply by kingjia90** [reply]
Upgrading [datahub above 1.3.2](https://github.com/pimcore/pimcore/discussions/12192#discussioncomment-2833473) didn't work out?

**Reply by Piotrulaa** [reply]
Unfortunately it didn't :(
I currently have datahub version 1.4.0 and problem still occurs.

**Comment by awiseturtle** [comment]
Could you try the following code: (without the TotalCount tag)
`{
  getProductListing(first: 10){
   edges{
    node{
      Title
    }
  }
}
}
`

**Reply by Piotrulaa** [reply]
Ok I tried, but unfortunately it didn't work too. Here is output:
```
{
  "data": {
    "getProductListing": {
      "edges": []
    }
  }
}
```

**Reply by awiseturtle** [reply]
Hi @Piotrulaa try this:
```
{
  getProductListing(first: 10){
   edges{
    node{
      Title: Title
    }
  }
}
}
```
Thanks

**Reply by Piotrulaa** [reply]
Thanks, I tried but output is the same as above :/

**Reply by Piotrulaa** [reply]
When I run 
```
{
  getProductListing(first: 10){
   edges{
     cursor
  }
}
}
```
The result is also
```
{
  "data": {
    "getProductListing": {
      "edges": []
    }
  }
}
```

Maybe it will help with tracking down a bit

**Reply by awiseturtle** [reply]
Hi @Piotrulaa 

Could you share your Datahub Query Schema, please?

thanks

**Comment by dvesh3** [comment]
@Piotrulaa please try [setting the error handling](https://pimcore.com/docs/data-hub/current/GraphQL/Configuration/Security_Settings.html#page_Error-Handling-Configuration-Values) to show errors instead of returning null and try again.

**Reply by Piotrulaa** [reply]
Good to know about this option, it didn't help but it set me in right direction to find solution by myself. I will post this solution in a moment.

**Comment by Piotrulaa** [comment, marked as answer]
I managed to figure out the source of problems.
In my test graphql config I had enabled option "Skip Permission Check" so I won't have to care about permissions while testing.
But I also added a row to field "Objects" in section "Workspaces" with permissions for all CRUD operations BUT I didn't fill "Path". It stayed empty.
The behaviour in such situation was as described - getObjectListings always returned an empty list while other queries worked as they should. Actually, isn't it a bug?

When I typed `/` into "Path" field Listings started to return list of Objects as they should :)

Thanks for all the efforts to everyone involved!

**Reply by dvesh3** [reply]
>I also added a row to field "Objects" in section "Workspaces" with permissions for all CRUD operations BUT I didn't fill "Path". It stayed empty.
> The behaviour in such situation was as described - getObjectListings always returned an empty list while other queries worked as they should. Actually, isn't it a bug?

yes, this sounds like a bug. could you please create an issue on [pimcore/data-hub](https://github.com/pimcore/data-hub/issues)? thanks!

**Reply by Piotrulaa** [reply]
Allright, I'll do it
